controllers/cruisecontroltask_controller.go

```diff
@@ -195,55 +309,71 @@ func (r *CruiseControlTaskReconciler) UpdateStatus(ctx context.Context, instance
 		return nil
 	}
 
-	err := r.Client.Status().Update(ctx, instance)
-	if err != nil {
+	if err := util.RetryOnConflict(util.DefaultBackOffForConflict, func() error {
+		log.Info("Updating status....")
+		err := r.Status().Update(ctx, instance)
 		if apiErrors.IsConflict(err) {
-			err = r.Client.Get(ctx, types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, instance)
+			err := r.Client.Get(ctx, types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, instance)
 			if err != nil {
 				return errors.WithMessage(err, "failed to get updated Kafka Cluster CR before updating its status")
 			}
 			taskAndStates.SyncState(instance)
-			err = r.Client.Status().Update(ctx, instance)
-			if err != nil {
-				return err
-			}
-		} else {
-			return err
 		}
+		return err
+	}); err != nil {
```

Comment:
**Recommended**: I think we **SHOULD NOT** put an inline function definition argumented function call into a short-statemented-if statement, because it makes everything much more difficult to read and thus maintain properly - the code logic/control flow is very difficult to understand at a glance and we don't really spare that much for it.

```golang
	conflictRetryFunction := func() error {
		log.Info("Updating status....")
		err := r.Status().Update(ctx, instance)
		if apiErrors.IsConflict(err) {
			err := r.Client.Get(ctx, types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, instance)
			if err != nil {
				return errors.WithMessage(err, "failed to get updated Kafka Cluster CR before updating its status")
			}
			taskAndStates.SyncState(instance)
		}
		return err
	}
	
	if err := util.RetryOnConflict(util.DefaultBackOffForConflict, conflictRetryFunction); err != nil {
		return err
	}
```

I know we are "wasting" the variable, but I do believe - not really in the mood to search for and prove - the Go compiler is smart enough to detect that if a variable is only used once in the function then its value can be inlined into the single usage place when generating assembly code. I do not want to trade readability for (almost certainly premature) optimization.

(Apologies, this is a big pet peeve of mine.)